src/lib.ts

```diff
@@ -83,6 +84,41 @@ const parseIntOpt = (int: string): Option<number> => {
 	return isNaN(parsed) ? none : some(parsed);
 };
 
+const resultMap3 = <A, B, C, D>(f: (a: A, b: B, c: C) => D) => <E>(
+	resultA: Result<E, A>,
+) => (resultB: Result<E, B>) => (resultC: Result<E, C>): Result<E, D> => {
+	if (resultA.kind === ResultKind.Err) {
+		return resultA;
+	}
+
+	if (resultB.kind === ResultKind.Err) {
+		return resultB;
+	}
+
+	if (resultC.kind === ResultKind.Err) {
+		return resultC;
+	}
+
+	return ok(f(resultA.value, resultB.value, resultC.value));
+};
+
+const resultMap2 = <A, B, C>(f: (a: A, b: B) => C) => <E>(
+	resultA: Result<E, A>,
+) => (resultB: Result<E, B>): Result<E, C> => {
+	if (resultA.kind === ResultKind.Err) {
+		return resultA;
+	}
+
+	if (resultB.kind === ResultKind.Err) {
+		return resultB;
+	}
+
+	return ok(f(resultA.value, resultB.value));
+};
+
+const fold = <A, B>(f: (value: A) => B, ifNone: B) => (opt: Option<A>): B => {
+	return withDefault(ifNone)(map(f)(opt));
+};
```

Comment:
This is nice 🚀 - similar signature to the `maybeRender` function above. Perhaps we should re-write that function to use this under the hood 🤔.